### DISCUSSION TITLE
Make your Issue stream more actionable with priority

### DISCUSSION THREAD

**Discussion by rachrwang** [getsentry/sentry#68908] [Upcoming Changes]
### Overview

Our aim at Sentry is to surface important issues for customers to review, triage, and fix. Given the volume of issues being created each day, it can be difficult to isolate critical issues from less important ones. Existing metadata, like log level, are helpful but imperfect tools for prioritization. A **fatal** issue might be less important to resolve than an **error**, if the latter is escalating in volume and impacting a significant number of users. 

As part of making Sentry more actionable, we’ve been investing in our ability to classify issues into different priority levels. Priority is determined by factors like log level, trends in error volume, error message, and historical actions taken on similar issues.

There are three priority levels for issues in Sentry:

- **High**: Issues that are likely to be actionable and require immediate attention.
- **Medium**: Issues that are likely to be actionable and require attention in the near future.
- **Low**: Issues that don't require immediate attention.

We’ll be releasing this functionality to [early adopters](https://docs.sentry.io/product/accounts/early-adopter-features/), starting the week of April 15. You can [learn more here.](https://docs.sentry.io/product/issues/issue-priority/)

### How will this impact your experience?

Sentry will be filtering out low priority issues from the default view of the issue stream. In addition, you can modify the priority of individual or multiple issues, which can then be used to search or filter on the stream. Over time, Sentry will be learning from actions taken to modify priority to improve the experience.

![image](https://github.com/getsentry/sentry/assets/82916311/78d961d8-9766-4311-b391-1bdfd7d98979)

![image](https://github.com/getsentry/sentry/assets/82916311/7bbba471-a1e1-47f1-b6a3-32b837afe10c)

### Have feedback for the Sentry team?

We’d love to learn more about how the new priority experience is working for you, and what we can do to make it even better. Some questions we have for our early adopter customers are:

1. Is your default issue stream more actionable with a priority filter? Why or why not?
2. Are there cases where you think priority is misclassified? Can you share examples?
3. Is modifying priority a useful part of your workflow?
4. Is there anything you’d want the Sentry team to change about the current implementation of priority?

**Comment by ulasozguler** [comment]
There should be a way to disable the default priority filter from the default view.

I have no intention of using this feature and I'd like to see every single issue, every time. This requires me to manually remove the filter now.

It does mislabel high-volume ongoing issues as low priority. Most examples I found are using a custom fingerprint like this:
```python
with sentry_sdk.push_scope() as scope:
    scope.fingerprint = [message]
    sentry_sdk.capture_message(message)
```

Regardless of specific examples, there should be a way to opt out of filtering out issues by default.

**Reply by julienlafont-shine** [reply]
I totally agree with that. The default filter hides some important issues on our side.

In additioj, I'd also prefer to disable automatic priority assignment and leave it to the developers. Sentry lacks tools to sort/categorize errors, this feature could help but not with this automation.

**Comment by deathlyrage** [comment]
This is a terrible feature, I want to disable this by default.

**Comment by SteenSchutt** [comment]
- Is your default issue stream more actionable with a priority filter? Why or why not?
Yes, as long as I am the one assigning priorities. When issues have priorities by default, especially the low priority, I end up missing them.

- Are there cases where you think priority is misclassified? Can you share examples?
I don't pay too much attention to how it classifies things, but I have found that it likes to set "undefined variable" warnings from PHP as low priority. Maybe this is just because it is in an old project with a PHP version where undefined variables are just notices, but even before they started making them emit warnings, I would not want them to be classified as low priority. Maybe it's just me, but I personally don't think any error is low priority.

- Is modifying priority a useful part of your workflow?
Maybe, but not as long as it happens automatically. As it stands right now I am just ignoring the feature to the best of my ability, because automatically classified issues are lumped in with the ones I have handled and classified myself. The Mark Reviewed feature could help, but for some inexplicable reason, this button, which you are presumably supposed to press on every issue, is hidden in the "..." menu. It would be much easier to triage if I could just filter for unprioritized issues.

- Is there anything you’d want the Sentry team to change about the current implementation of priority?
I want to be able to disable automatic prioritization entirely, or at the very least the default priority filter. It is beyond frustrating to have to jump back and forth between filters just to get an idea about what's happening. I'll filter out things myself if I need to, or create saved searches for that. It certainly doesn't help that projects from all of my teams are lumped together now, so my projects list is just 50 ungrouped entries I'll have to look through.

I'm sure this is useful to a lot of people, but it is incredibly frustrating for me. In my opinion it needs to be possible to remove the default filter, disable the automatic prioritization, remove the priority from an issue (So it doesn't have any), and potentially tweak the rules used to prioritize the issues.
